client/src/components/BreadCrumb.tsx

```diff
@@ -7,6 +7,8 @@ import { RootState } from "../reducers/index";
 import { connect } from "react-redux";
 import { Dispatch } from "redux";
 
+var latestState = 0;
```

Comment:
We shouldn't store global variables, if we want to maintain state, we should do it in the component. Look at Courses component to see how to store state in function components (ie: useState) .